core/distributed/helpers.hpp

```diff
@@ -168,6 +172,33 @@ void vector_dispatch(T* linop, F&& f, Args&&... args)
 }
 
 
+/**
+ * Returns the local vector as a LinOp
+ * @tparam LinOpType either LinOp or const LinOp
+ * @param op  the object from which the local vector is extracted
+ * @return  the local vector
+ */
+template <typename LinOpType, typename = std::enable_if_t<std::is_same<
+                                  LinOp, std::decay_t<LinOpType>>::value>>
+LinOpType* get_local(LinOpType* op)
+{
+    LinOpType* local = nullptr;
+#if GINKGO_BUILD_MPI
+    if (is_distributed(op)) {
+        run<gko::experimental::distributed::Vector, float, double,
+            std::complex<float>, std::complex<double>>(
+            op, [&](auto vector_op) { local = get_local(vector_op); });
+    } else
+#endif
+    {
+        run<gko::matrix::Dense, float, double, std::complex<float>,
+            std::complex<double>>(
+            op, [&](auto vector_op) { local = get_local(vector_op); });
```

Comment:
I think the nomenclature is somewhat confusing here. What is the meaning of `get_local`  in the non-distributed case? I always thought get_local to be rank local vector part.